cluster-autoscaler/core/scaledown/planner/planner.go

```diff
@@ -289,15 +296,34 @@ func (p *Planner) categorizeNodes(podDestinations map[string]bool, scaleDownCand
 		// TODO(x13n): break on timeout. Figure out how to handle nodes
 		// identified as unneeded in previous iteration, but now
 		// skipped due to timeout.
-		removable, unremovable := p.rs.SimulateNodeRemoval(node, podDestinations, p.latestUpdate, pdbs)
-		if unremovable != nil {
-			unremovableCount += 1
-			p.unremovableNodes.AddTimeout(unremovable, unremovableTimeout)
-		}
+		removable, unremovable := p.rs.SimulateNodeRemoval(node, podDestinations, p.latestUpdate, p.pdbRemainingDisruptions.GetPdbs())
 		if removable != nil {
+			ok, isRisky, blockingPod := p.pdbRemainingDisruptions.CanDisrupt(removable.PodsToReschedule)
+			if !ok {
+				if blockingPod != nil {
+					unremovable = &simulator.UnremovableNode{
+						Node:        removable.Node,
+						Reason:      simulator.BlockedByPod,
+						BlockingPod: blockingPod,
+					}
+				}
+				continue
+			}
+			if isRisky {
+				removable.IsRisky = true
+			}
 			delete(podDestinations, removable.Node.Name)
+			err := p.pdbRemainingDisruptions.Update(removable.PodsToReschedule)
```

Comment:
`Update` shouldn't really be able to return an error now. Building a selector for each pdb can be done once during `pdbRemainingDisruptions` creation, so the error is handled there. We don't need to build the same selector twice per node anyway (once in `CanDisrupt`, then again in `Update`).